### ISSUE TITLE
importing tools/libraries overwrites custom block with same name

### ISSUE THREAD

**Issue by maryfries** [jmoenig/Snap--Build-Your-Own-Blocks#1303]
We don't have a proposed solution for this, but it is alarming to have your block overwritten with no notice of any kind.

**Comment by cycomachead** [comment]
What library block did this happen to?

Certainly, I can see where there are confusing cases from this behavior, but they seem to be rather rare. I definitely know that students (and TAs have even recommending to) use this replacement policy as a way of "recovering" or "resetting" custom blocks which have been tinkered.

**Comment by jmoenig** [comment]
Yes, I'm using this feature myself a lot, e.g. to overload the recursive ```map``` implementation with a "fast" one that's using JS, or when fixing a bug in a library.

It's a little funny that this is considered an issue, because it was actually a rather big effort for me to enable overwriting one custom block with another. Snap can totally deal with different versions of the same block signature having different definitions, but boy, *that* really gets confusing. That said, I'm working on polymorphism for blocks and with that there is probably going to be some kind of module system. The details are still fuzzy, though...

**Comment by liam4** [comment]
Could a dialog containing a list of checkmarks next to blocks that asks which blocks to overwrite work well here? (And maybe give the user to view the implementation of the overwriting blocks, at the same time!)

**Comment by jmoenig** [comment]
No. Michael is right, we need to know the particular case here. Namespace is a side-effect of a text-based programming language, and a feature in a blocks-based one.
